src/PrMerger.js

```diff
@@ -43,12 +44,31 @@ class PrMerger {
     // there is a PR still-in-merge.
     async runStep() {
         Logger.info("runStep running");
-        const currentContext = await this._current();
-        if (currentContext && !(await this._finishContext(currentContext)))
```

Comment:
Does _finishContext() _start_ something long/expensive? If it only makes a few GitHub calls without initiating anything, then I think it is OK to let it run here. We just cannot `return true` if the currentContext is unlabeled and there are labeled contexts now. On other other hand, we do not really want to `return true` in any case because other PRs may need some processing too, even of all of them are unlabeled.

Reply:
> Does _finishContext() start something long/expensive?

The bot does not have expensive methods: GitHub calls are not expensive AFAICT. This method (as well as another, startProcessing()), initiate several (or several dozens) API calls (taking several seconds).
So, I can try to follow this suggestion (simplifying the code at the expense of optimization).

Reply:
> Does _finishContext() start something long/expensive?

For example, something that may trigger new CI tests.